### PR TITLE
Add support for QZSS constellation [OI-1792]

### PR DESCRIPTION
**Pull request by cstolowicz**
# Description

@swift-nav/devinfra

- Added the `use_qzss_sat` field to the High Level Flag Message which decreased the amount of reserved fields from 6 to 5.
- The Validation workflow will be disabled before merging this PR. The reasons about this decision can be found [here](https://snav.slack.com/archives/C03V0SS40/p1720119701332469?thread_ts=1720096667.884599&cid=C03V0SS40)

# API compatibility

Does this change introduce a API compatibility risk?

## API compatibility plan

# JIRA Reference

https://swift-nav.atlassian.net/browse/OI-1792

### CODE REVIEW

File: .github/workflows/c.yaml [hunk 1]
```diff
@@ -31,6 +31,8 @@ jobs:
 
     runs-on: ubuntu-latest
     container: ubuntu:18.04
+    env:
+      ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
```
Comment (cstolowicz) [inline .github/workflows/c.yaml:35]: This was implemented in order to solve an error on the workflow. Discussion can be found [here](https://snav.slack.com/archives/C02E0A1CE3Z/p1720180486047209)

File: spec/yaml/swiftnav/sbp/gnss.yaml [hunk 2]
```diff
@@ -45,6 +45,9 @@ definitions:
                       - 13: BDS2 B2
                       - 14: GAL E1B
                       - 20: GAL E7I
+                      - 31: QZS L1CA
```
Comment (cstolowicz) [inline spec/yaml/swiftnav/sbp/gnss.yaml:48]: Referring to all the constellations here (not QZSS): Is there a reason why only these signal codes were added and not all of them?

File: proto/integrity.proto [hunk 3]
```diff
@@ -73,10 +73,11 @@ message MsgSsrFlagHighLevel {
     uint32 use_gps_sat = 7;
     uint32 use_gal_sat = 8;
     uint32 use_bds_sat = 9;
-    uint32 use_tropo_grid_points = 11;
-    uint32 use_iono_grid_points = 12;
-    uint32 use_iono_tile_sat_los = 13;
-    uint32 use_iono_grid_point_sat_los = 14;
+    uint32 use_qzss_sat = 10;
+    uint32 use_tropo_grid_points = 12;
```
[the remark this answers is not in the document]
Comment (cstolowicz) [reply proto/integrity.proto:77]: We've discussed about changing the reserved field array for five different reserved fields but each field should have a different name. We've also discussed about deprecating the previous high level message. I believe these changes entail a more profound thought about how to design this new message, which may be outside of the scope of my task.
Wdyt about leaving it as I defined it and creating a ticket to revisit the way the reserved fields are defined, on this and other messages as well?

File: spec/tests/yaml/swiftnav/sbp/integrity/test_MsgSsrFlagHighLevel.yaml [hunk 4]
```diff
@@ -18,13 +18,13 @@ tests:
       use_gps_sat: 1
       use_gal_sat: 2
       use_bds_sat: 3
+      use_qzss_sat: 0
```
[the remark this answers is not in the document]
Comment (cstolowicz) [reply spec/tests/yaml/swiftnav/sbp/integrity/test_MsgSsrFlagHighLevel.yaml:21]: I'll be able to tackle this change on Wednesday. I've been trying to obtain the CRC with the change and I've been having some issues with it, that's why I haven't uploaded the change yet

**Review by woodfell — APPROVED** [review]

**Review by pcrumley — COMMENTED** [review]
it's hard for me to approve this without having a little more context. I can see where you are coming from with disabling the message validation CI check. We should be sure to re-enable it after you merge this PR

I believe the Benchmark CI check is just flat out broken because no runners are picking up the job

**Comment by cstolowicz** [comment]
> it's hard for me to approve this without having a little more context. I can see where you are coming from with disabling the message validation CI check. We should be sure to re-enable it after you merge this PR
> 
> I believe the Benchmark CI check is just flat out broken because no runners are picking up the job

@pcrumley To give you more context:

We're working on adding integrity support for QZSS and that involves changing the high level flag messages to also report the status of the QZSS satellites. The new QZSS field will replace one of the reserved fields as stated on the [ICD](https://docs.google.com/spreadsheets/d/1Vezr7LlK-TdbqOHHCQa4OQ35roZg0vBQ/edit?gid=1884888174#gid=1884888174) and coded [here](https://github.com/swift-nav/libsbp/blob/073233befe20c8742f7f766db873143390b316d7/spec/yaml/swiftnav/sbp/integrity.yaml#L133).

Feel free to ask any questions if needed.

**Comment by cstolowicz** [comment]
> There is no runner that is going to pick-up this job https://github.com/swift-nav/libsbp/actions/runs/9876111694/job/27274391320?pr=1414 `macos-11` is no longer a [valid label](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for--private-repositories)

Yes @fpezzinosn , sorry, I forgot to merge against your [branch](https://github.com/swift-nav/libsbp/pull/1417#discussion_r1672235723)

**Review by pcrumley — APPROVED** [review]

**Review by valeparrags — APPROVED** [review]
Reviewed: Add support for QZSS constellation commit
LGTM!

**Pull request merged**